src/main.rs

```diff
@@ -239,25 +305,61 @@ fn cargo_careful(mut args: env::Args) {
             ),
     };
 
+    let mut san_to_try = None;
+
     // Go through the args to figure out what is for cargo and what is for us.
     let mut cargo_args = Vec::new();
     for arg in args.by_ref() {
-        if let Some(_careful_arg) = arg.strip_prefix("-Zcareful-") {
-            // A flag for us! So far we don't support any, though.
-            show_error!("unsupported careful flag `{}`", arg);
+        if let Some(careful_arg) = arg.strip_prefix("-Zcareful-") {
+            let (key, value): (&str, Option<&str>) = match careful_arg.split_once('=') {
+                Some((key, value)) => (key, Some(value)),
+                None => (careful_arg, None),
+            };
+            match (key, value) {
+                ("sanitizer", Some(san)) => san_to_try = Some(san.to_owned()),
+                ("sanitizer", None) => san_to_try = Some(DEFAULT_SANITIZER.to_owned()),
+                _ => show_error!("unsupported careful flag `{}`", arg),
+            }
+            continue;
         } else if arg == "--" {
             // The rest is definitely not for us.
             break;
         }
+
         // Forward regular argument.
         cargo_args.push(arg);
     }
     // The rest is for cargo to forward to the binary / test runner.
     cargo_args.push("--".into());
     cargo_args.extend(args);
 
+    let sanitizer = san_to_try.and_then(|san| {
+        sanitizer(&target, &san).map_or_else(
+            |e| {
+                eprintln!("{e}\nFailed to get list supported sanitizers! Falling back to none.");
+                None
+            },
+            |b| {
+                if b {
+                    eprintln!(
+                        "Using sanitizier `{san}`."
+                    );
+                    Some(san)
+                } else {
+                    eprintln!("Sanitizer `{san}` not supported by target, falling back to none.");
```

Comment:
Wouldn't it be better to bail out here? Seems like an error if the user asks for a sanitizer that is not supported.